debezium-core/src/main/java/io/debezium/relational/history/KafkaDatabaseHistory.java

```diff
@@ -128,11 +147,21 @@
     private volatile KafkaProducer<String, String> producer;
     private int maxRecoveryAttempts;
     private Duration pollInterval;
+    private final ExecutorService checkTopicSettingsExecutor;
+
+    public KafkaDatabaseHistory() {
+        checkTopicSettingsExecutor = Executors.newSingleThreadExecutor((r) -> {
```

Comment:
Why not using our own `Threads` factory? It takes care of the naming and should be the one facility for spawning new threads.